src/engine.rs

```diff
@@ -165,6 +166,9 @@ where
                     };
                     writer.set_output(res);
                 }
+                debug_assert!(
+                    file_context.id == self.pipe_log.fetch_active_file(LogQueue::Append)?.id
+                );
```

Comment:
Let's change the `Engine::sync` to call `self.write(&mut LogBatch::default(), true)` instead of directly calling pipe method. Or this assertion might fail when there's concurrent write and sync.